### DISCUSSION TITLE
Request: Support ambient lighting?

### DISCUSSION THREAD

**Discussion by ReplicaObscura** [jseidl/magic-areas#225] [Ideas / Feature Requests]
Hello,

Mostly for my outdoor lights such as my porch and driveway, I'm trying to figure out if there's a way to combine Magic Areas with slowly-adjusting ambient lighting.

On your suggestion, I'm now using Adaptive Lighting along with Magic Areas to help set the brightness of the lights to an appropriate level when they turn on, and that works pretty well.

However, here's what I would really like to happen with my outdoor lights, and I'm not sure if it's possible:

- During the day, they are always off
- As the sun goes down, the outdoor lights turn on very dimly, and slowly increase in brightness up to about 50%
- When motion is detected, they go to a higher brightness level
- When the motion is cleared, they go back to their normal ambient level
- As the sun rises, the ambient level gets dimmer and eventually shuts off for the day

The slowly changing brightness I could live without--simply having them come on at 50% during the night and then get brighter when motion is detected would be a reasonable compromise as well.

I know this can be done with a complex set of automations, but I would love to find a way to still use Magic Areas in those areas and have it behave similarly to this.

Is there any way something like this is currently possible, or could ambient lighting options perhaps be added in the future? If I can think of a good way to accomplish it I could submit a PR, but thus far I'm not even sure what the best approach would be considering Magic Areas isn't concerned with the brightness of the lights.

Thanks!

**Comment by jseidl** [comment]
> During the day, they are always off
Currently this is possible by using either an outdoor luminance sensor or using the `sun` sensor as your `dark` sensor.

Are you using the 2.x release or the 3.x beta?

Currently there's no support for having lights on when an area isn't occupied, but I get the use case, it would be interesting to have but currently MA doesn't supports it yet. 

Thanks for the suggestion, might add in the future.

The brightness stuff you can tweak the sunrise and sunset offset along with min and max brightness on Adaptive Lighting. For the dim states, you can simulate with the sleep mode on Adaptive Lighting as well (as a temporary workaround)
